adding-spring-security/leaveapp-complete/src/test/java/com/jdriven/leaverequest/LeaveRequestControllerSpringBootWebEnvRandomPortTest.java

```diff
@@ -2,57 +2,68 @@
-	private static final String TOKEN_HR_VALID_UNTIL_AUGUST_2022 = "eyJhbGciOiJSUzI1NiIsInR5cCIgOiAiSldUIiwia2lkIiA6ICJRSS1mdW1sLTRyTnBrdVZxWVg0elpuZlRUZW1hSkxoZ183Z0dULTZiSFlVIn0.eyJqdGkiOiJhZjJkYjUzZi0zMzA3LTRiYmItOWIyMy01NjI4NzYxOTNiOTAiLCJleHAiOjE2NTk4OTgyMjIsIm5iZiI6MCwiaWF0IjoxNTczNDk4MjIyLCJpc3MiOiJodHRwOi8vbG9jYWxob3N0OjgwOTAvYXV0aC9yZWFsbXMvc3ByaW5nLWNsb3VkLWdhdGV3YXktcmVhbG0iLCJhdWQiOiJhY2NvdW50Iiwic3ViIjoiYjc2ODI2NjQtZWJhZC00NzAzLTgyNTctNDIyY2QwNTJhNjU4IiwidHlwIjoiQmVhcmVyIiwiYXpwIjoic3ByaW5nLWNsb3VkLWdhdGV3YXktY2xpZW50Iiwibm9uY2UiOiJDMFktek5kTUt6UWwweGdESDc1OWZJOU13ek5iY3ROT3I2M0ZOMUFSX1NzIiwiYXV0aF90aW1lIjoxNTczNDk4MjIyLCJzZXNzaW9uX3N0YXRlIjoiODA5NGMxZGUtMTNmYS00MzQzLWEyN2MtODRhZTE3NDJlMDA5IiwiYWNyIjoiMSIsInJlYWxtX2FjY2VzcyI6eyJyb2xlcyI6WyJvZmZsaW5lX2FjY2VzcyIsIkhSIiwidW1hX2F1dGhvcml6YXRpb24iXX0sInJlc291cmNlX2FjY2VzcyI6eyJhY2NvdW50Ijp7InJvbGVzIjpbIm1hbmFnZS1hY2NvdW50IiwibWFuYWdlLWFjY291bnQtbGlua3MiLCJ2aWV3LXByb2ZpbGUiXX19LCJzY29wZSI6Im9wZW5pZCBwaG9uZSBtaWNyb3Byb2ZpbGUtand0IG9mZmxpbmVfYWNjZXNzIHByb2ZpbGUgZW1haWwgYWRkcmVzcyIsInVwbiI6ImJvYiIsImVtYWlsX3ZlcmlmaWVkIjp0cnVlLCJhZGRyZXNzIjp7fSwibmFtZSI6IkJvYiIsImdyb3VwcyI6WyJvZmZsaW5lX2FjY2VzcyIsIkhSIiwidW1hX2F1dGhvcml6YXRpb24iXSwicHJlZmVycmVkX3VzZXJuYW1lIjoiYm9iIiwiZ2l2ZW5fbmFtZSI6IkJvYiJ9.DxqIEqanm_7pM08amW6VQ7EBXta54c7kV2ozewYZ_K6_nqOZHAQMWiH-DTnCLlNPnoKZeoLsf_7QcRAXijGoCiRYy0suhGvLAAMCtQgDMfmgMYCZV-sOkHNX8AZDzDj3tXQs1zVQOc3zw9M3Byt_e-0KzxcIi4EUcD8p5fi7NTsynV7Lm6KQVMQTVE-eWBN9wtNo6Lacv4WeFcOTYTRTVRHJU72rrXe6hSDpRd90__T2PsncnqvKPhtaSe4klgaENflxQRTEVKu_oDah4MabwFW1yqQRX68OvUdmtOI2sd05p1juYwuIbSThGtcgrqA1VomgHu0fbp6bgvd1Q_neuw";
-
 	@Autowired
 	private LeaveRequestRepository repository;
 
 	@Autowired
-	private TestRestTemplate resttemplate;
+	private TestRestTemplate restTemplate;
+
+	@MockBean
+	private JwtDecoder jwtDecoder;
 
 	@BeforeEach
 	void beforeEach() {
 		repository.clear();
+		when(jwtDecoder.decode(anyString())).thenReturn(Jwt.withTokenValue("token")
+				.subject("Alice")
+				.claim("sub", "user")
+				.claim("scope", "read")
+				.claim("realm_access", Map.of("roles", List.of("HR")))
```

Comment:
Originally these tests had separate tokens for Alice and HR, to correctly show authorization based on roles.

Let's introduce the same Nested test classes as seen in the initial sample, and then add separate mocking for either Alice or HR.

Reply:
Fixed in https://github.com/timtebeek/spring-security-samples/pull/103